src/target/macos.rs

```diff
@@ -19,6 +26,53 @@ use std::{
 // For more information about ASLR, you can refer to https://en.wikipedia.org/wiki/Address_space_layout_randomization
 const _POSIX_SPAWN_DISABLE_ASLR: i32 = 0x0100;
 
+// Max number of characters to read from a thread name.
+const MAX_THREAD_NAME: usize = 100;
+
+struct OSXThread {
+    port: mach_port_name_t,
+    pthread_id: Option<usize>,
+    task_port: ipc_space_t,
+}
+
+impl Drop for OSXThread {
+    fn drop(&mut self) {
+        let result = unsafe { mach::mach_port::mach_port_deallocate(self.task_port, self.port) };
+        if result != kern_return::KERN_SUCCESS {
+            panic!("Failed to deallocate port!");
+        }
+    }
+}
+
+extern "C" {
+    // FIXME: Use libc  > 0.2.74 when available
+    pub fn pthread_from_mach_thread_np(port: libc::c_uint) -> libc::pthread_t;
+}
+
+impl Thread for OSXThread {
+    type ThreadId = mach_port_t;
+
+    fn name(&self) -> Result<Option<String>, Box<dyn Error>> {
+        if let Some(pt_id) = self.pthread_id {
+            let mut name = [0 as libc::c_char; MAX_THREAD_NAME];
+            let name_ptr = &mut name as *mut [libc::c_char] as *mut libc::c_char;
+            let get_name = unsafe { libc::pthread_getname_np(pt_id, name_ptr, MAX_THREAD_NAME) };
+            if get_name == 0 {
+                let opt_name = unsafe { CStr::from_ptr(name_ptr) }
+                    .to_str()
+                    .map(|s| s.to_owned())
+                    .ok();
+                return Ok(opt_name);
+            }
+        };
+        Ok(None)
```

Comment:
When there is no pthread existing for the task this should indeed return `Ok(None)`.